[wasm] Misc CI/test run improvements

radical opened the pull request:
- Update chrome+chromedriver for helix tests to latest stable (`96.0.4664.45`)
- Enable library tests to run on Windows, for manual runs

- Disable windows/AOT builds completely due to https://github.com/dotnet/runtime/issues/61721
- Disable failing WBT tests on windows due to https://github.com/dotnet/runtime/issues/61725
- Disable JS interop tests failing with new chrome - https://github.com/dotnet/runtime/issues/61945

radical committed:
[wasm] Disable WBT tests failing on windows

https://github.com/dotnet/runtime/issues/61725

radical committed:
[wasm] Update chrome used for testing to latest stable

Latest stable is `96.0.4664.45`, with `branch_base_position=929512`, and
`929513` is the closest available build.

radical committed:
[wasm][aot] Don't try to run the tests, if AOT build fails

radical committed:
[wasm] Disable AOT builds on windows

Issue: https://github.com/dotnet/runtime/issues/61721

radical commented:
/azp run runtime-manual

radical committed:
[wasm] Disable JS Interop failing tests

https://github.com/dotnet/runtime/issues/61945

radical committed:
[wasm] Enable windows tests on manual runs

radical committed:
[wasm] honor RunSmokeTestsOnly

lewing reviewed (approved):
The test exclusions and chrome bump look good, @steveisok can you take a look at the pipeline bits

radekdoulik commented on src/libraries/sendtohelixhelp.proj:
```diff
@@ -364,15 +364,20 @@
     </PropertyGroup>
+
+    <!-- Version number to revision number mapping from http://omahaproxy.appspot.com/ and find the closest one in
+         https://commondatastorage.googleapis.com/chromium-browser-snapshots/index.html?prefix=Linux_x64/
+
+         Eg. latest stable version is 96.0.4664.45 with revision 929512.
+         but the closest one available in the snapshosts is 929513
```
```suggestion
         https://commondatastorage.googleapis.com/chromium-browser-snapshots/index.html?prefix=Linux_x64/
         and https://commondatastorage.googleapis.com/chromium-browser-snapshots/index.html?prefix=Win_x64/

         Eg. latest stable version is 96.0.4664.45 with revision 929512.
         but the closest one available in the snapshosts is 929513.
         Please make sure to check both platforms as sometime
         the same snapshot might not be available in one of them.
```

radical committed:
Update src/libraries/sendtohelixhelp.proj

Co-authored-by: Radek Doulik <radek.doulik@gmail.com>

radical committed:
[wasm][win] fix wasm/windows networking test failures

`System.Net.WebSockets.Client.Tests`, and
`System.Net.Http.Functional.Tests` fail because the echo servers don't
get included in the helix payload.

 - Don't condition the inclusion on `RunSmokeTestsOnly`, instead it will
   be included only if it was built, and that would depend correctly on
   `RunSmokeTestsOnly`.
- Pass the smoke test argument to the build command line for
  wasm/windows

radical committed:
Merge remote-tracking branch 'origin/main' into misc-wasm-fixes

radical committed:
Add condition to ensure that the middleware exists

radical commented:
/azp run runtime-manual

steveisok reviewed (approved):
LGTM
